### ISSUE TITLE
While fetching commits, we should parse all branches and remove duplicates.

### ISSUE THREAD

**Issue by gautamrege** [joshsoftware/code-curiosity#6]
We currently fetch commits for the team only from master.
* We should fetch commits from all the branches
* Remove duplicates / merges.

Alternatively, we will have to ask the team to tell us which branch commits to parse but that could have it's own set of problems. 

[From Rishi's email]
It is ideally suggested while working on open-source projects, you should keep the remote master and your local master copy in sync. So you should not merge your new feature branch into local master. Instead wait for the author to accept your PR and then you can pull from remotes master into local master.

If you follow this process, it helps in long term too. Now imagine one member of team alpha (say Rishi), he has done some changes on the local master(instead of new feature branch), and sends a PR. But it got rejected by the main author (Jiren), so now he has a master which has some commits which is not acceptable. So what he can do now(because he do not need those faulty commits):

1. git reset the commits and then checkout.
2. Delete the repo and re-clone it.

or it could get worst, if Rishi does not work on filterjs for some months and then he comes back, so Rishi will just pull the remotes master into local master(which also has those unaccepted commits) to get the latest copy. Rishi may not remember that there are few commits which were not acceptable. And now he has a master which is not true copy of remote master and it might screw up his future work.

I am going to put this in Wiki as guidelines to contribute to Open Source!

**Comment by kirandc** [comment]
@gautamrege We are fetching all branches commits. I already added uniqueness on commit message and date.

**Comment by gautamrege** [comment]
ok. interesting .. @narutosanjiv committed to rails32 branch of joshsoftware/mailserv but it did not get picked up. Please check that and fix - it could be some issue from his account too. Once that is done, please close this ticket.
